var/spack/repos/builtin/packages/octopus/package.py

```diff
@@ -39,13 +39,13 @@ class Octopus(AutotoolsPackage, CudaPackage):
     version("develop", branch="main")
 
     variant("mpi", default=True, description="Build with MPI support")
-    variant("scalapack", default=False, description="Compile with Scalapack")
+    variant("scalapack", default=False, description="Compile with Scalapack", when="+mpi")
     variant("metis", default=False, description="Compile with METIS")
-    variant("parmetis", default=False, description="Compile with ParMETIS")
+    variant("parmetis", default=False, description="Compile with ParMETIS", when="+mpi")
     variant("netcdf", default=False, description="Compile with Netcdf")
     variant("arpack", default=False, description="Compile with ARPACK")
     variant("cgal", default=False, description="Compile with CGAL library support")
-    variant("pfft", default=False, description="Compile with PFFT")
+    variant("pfft", default=False, description="Compile with PFFT", when="+mpi")
```

Comment:
Can I convince you to move the `when` args to after the `default`?  It's a bit more readable IMO.

Reply:
Sure! :)

Reply:
Thanks.